### ISSUE TITLE
GYM fails to package - Toolchain mismatch?

### ISSUE THREAD

**Issue by arunks** [fastlane/fastlane#6003]
### New Issue Checklist
- [x] Updated fastlane to the latest version - 1.8.0
- [x] I have read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
### Issue Description

When running gym with xcode8, and toolchain set to 
 --toolchain com.apple.dt.toolchain.Swift_2_3

The app is built fine and it gets to signing and packaging. Here when its trying to include the swift libraries, it fails by referencing the - XcodeDefault.xctoolchain

For some reason the toolchain the packager looks at is the Default toolchain eventhough I reference the Swift_2_3 toolchain in gym.
##### Complete output when running fastlane, including the stack trace and command used

```
Found no similar issues. To create a new issue, please visit:
https://github.com/fastlane/fastlane/issues/new
/System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/fileutils.rb:1380:in `initialize': [!] No such file or directory - /Applications/Xcode8.app/Contents/Developer//Toolchains/XcodeDefault.xctoolchain/usr/lib/swift/iphoneos/libswiftLocalAuthentication.dylib (Errno::ENOENT)
```
### Configuration Files

Please copy the complete content of your `Fastfile` and any other configuration files you use below:

Fastfile: (I cannot pate my whole file here but here is the gym option I am passing)
I am trying to run this commandline first before I formalize the fastfile.

``` ruby
 --toolchain com.apple.dt.toolchain.Swift_2_3
```
### Environment

fastlane version (run `fastlane -v`): 1.102.0

Do you use bundler to execute fastlane (i.e. `bundle exec fastlane`)? NO

Do you use a Ruby environment manager (e.g. `chruby`, `rbenv`, `rvm`)? NO

**Comment by nafu** [comment]
@arunks 

> Fastfile: (I cannot pate my whole file here but here is the gym option I am passing)
> I am trying to run this commandline first before I formalize the fastfile.

If you could paste gym action, it's very useful for us to help the problem you're facing.

Here is some tips that might help you.
Let me know whether these works for you! Thanks.

## Toolchain option usage

In order to use toolchain option, need to set `use_legacy_build_api: false`.

e.x.

```ruby
gym(
  workspace: WORKSPACE,
  scheme: SCHEME,
  toolchain: "com.apple.dt.toolchain.Swift_2_3",
  configuration: "AdHoc",
  use_legacy_build_api: false,
  export_method: "ad-hoc"
)
```

## Sample woking project

Here is the sample that works with Xcode 8 (Swift 2.3).
Check this out!

https://github.com/nafu/FastlaneWebKitFailure/tree/xcode-8-with-swift2.3

## Related Issue

https://github.com/fastlane/fastlane/issues/5998

**Comment by TKBurner** [comment]
@arunks Can you share your full input and output? I want to take a look at what is going on.

Thanks!

**Comment by arunks** [comment]
Thanks for the reply @nafu 

I am running GYM command line and have not integrated into the Fastfile yet. Here are the options I am using - 

```gym -c -w [***].xcworkspace -m -z -j enterprise -q "Release-Enterprise" -s [***] -y fastlane/xcconfig/debug.xcconfig --buildlog_path Build --use_legacy_build_api --toolchain com.apple.dt.toolchain.Swift_2_3 ```

The xcconfig just has a build_configuration_directory and code_signing_identity.
When I try turing off the --use_legacy_api I run into the radar issue. 

I am not sure how to force gym to use the Swift_2_3 toolchain for packaging.

**Comment by arunks** [comment]
@TKBurner are you looking for more information?

**Comment by thasegaw** [comment]
I am having similar issue. I think it is caused by this line. 
https://github.com/fastlane/fastlane/blob/master/gym/lib/gym/xcodebuild_fixes/swift_fix.rb#L30

In spite of the --toolchain is set to 'com.apple.dt.toolchain.Swift_2_3', the default toolchain folder is used.

**Comment by arunks** [comment]
@thasegaw Yes, I am working with @KrauseFx to address this issue in PR https://github.com/fastlane/fastlane/pull/6018

**Comment by jalmaas** [comment]
The toolchainpath contains double slash after `Developer`

**Comment by TKBurner** [comment]
@jalmaas Can you show me the full output (and input of the lane) where you are seeing this double slash? I have circled back with the team, and want to make sure we have as much info as possible on this thread for them to look at. Also, if you can give exact repro steps that'll help a lot.

**Comment by TKBurner** [comment]
@jalmaas We are working on implementing a fix! Check out #6018 to follow along.
